src/main/java/org/jenkinsci/plugins/ghprb/extensions/status/GhprbSimpleStatus.java

```diff
@@ -165,6 +174,11 @@ public void onBuildStart(AbstractBuild<?, ?> build,
         } else {
             sb.append(Ghprb.replaceMacros(build, listener, startedStatus));
         }
+
+        if(showMatrixStatus && !(build.getProject() instanceof MatrixProject)){
```

Comment:
If this is a MatrixProject then the if statement will never be entered, and only regular jobs will be affected.  You want
`if (build.getProject() instanceof MatrixProject && !showMatrixStatus) {`  or
`if (!showMatrixStatus && build.getProject() instanceof MatrixProject) {`